zilliqa/tests/it/eth.rs

```diff
@@ -493,22 +498,128 @@ async fn get_storage_at(mut network: Network) {
+    let tx = wallet.get_transaction(hash).await.unwrap().unwrap();
+    let receipt = wallet.get_transaction_receipt(hash).await.unwrap().unwrap();
+
+    (tx, receipt)
+}
+
+#[zilliqa_macros::test]
+async fn send_legacy_transaction(mut network: Network) {
+    let to = H160::random_using(network.rng.lock().unwrap().deref_mut());
+    let tx = TransactionRequest::pay(to, 123).into();
+    let (tx, receipt) = send_transaction(&mut network, tx).await;
+
+    assert_eq!(tx.transaction_type.unwrap().as_u64(), 0);
+    assert_eq!(receipt.to.unwrap(), to);
+}
+
+#[zilliqa_macros::test]
+async fn send_eip2930_transaction(mut network: Network) {
+    let (to, access_list) = {
+        let mut rng = network.rng.lock().unwrap();
+        let to = H160::random_using(rng.deref_mut());
+        let access_list = AccessList(vec![AccessListItem {
+            address: H160::random_using(rng.deref_mut()),
+            storage_keys: vec![
+                H256::random_using(rng.deref_mut()),
+                H256::random_using(rng.deref_mut()),
+            ],
+        }]);
+        (to, access_list)
+    };
+    let tx = Eip2930TransactionRequest::new(TransactionRequest::pay(to, 123), access_list.clone())
+        .into();
+    let (tx, receipt) = send_transaction(&mut network, tx).await;
+
+    assert_eq!(tx.transaction_type.unwrap().as_u64(), 1);
+    assert_eq!(tx.access_list.unwrap(), access_list);
+    assert_eq!(receipt.to.unwrap(), to);
+}
+
+#[zilliqa_macros::test]
+async fn send_eip1559_transaction(mut network: Network) {
+    let (to, access_list) = {
+        let mut rng = network.rng.lock().unwrap();
+        let to = H160::random_using(rng.deref_mut());
+        let access_list = AccessList(vec![AccessListItem {
+            address: H160::random_using(rng.deref_mut()),
+            storage_keys: vec![
+                H256::random_using(rng.deref_mut()),
+                H256::random_using(rng.deref_mut()),
+            ],
+        }]);
+        (to, access_list)
+    };
+    let tx = Eip1559TransactionRequest::new()
+        .to(to)
+        .value(456)
+        .access_list(access_list.clone())
+        .max_fee_per_gas(5000)
+        .max_priority_fee_per_gas(5000)
+        .into();
+    let (tx, receipt) = send_transaction(&mut network, tx).await;
+
+    assert_eq!(tx.transaction_type.unwrap().as_u64(), 2);
+    assert_eq!(tx.access_list.unwrap(), access_list);
+    assert_eq!(tx.max_fee_per_gas.unwrap().as_u64(), 5000);
+    assert_eq!(tx.max_priority_fee_per_gas.unwrap().as_u64(), 5000);
+    assert_eq!(receipt.to.unwrap(), to);
+}
+
+/// Test which sends a legacy transaction, without the replay protection specified by EIP-155.
+#[zilliqa_macros::test]
+async fn send_legacy_transaction_without_chain_id(mut network: Network) {
+    let wallet = network.genesis_wallet().await;
+
+    let tx = TransactionRequest::pay(
+        H160::random_using(network.rng.lock().unwrap().deref_mut()),
```

Comment:
Yeah. I thought it was particularly important here, rather than using a fixed address, so that we calculate and verify the expected hashes of multiple variants of each transaction type.